### PR TITLE
package.json: Update xterm

### PR DESCRIPTION
**Pull request by cockpituous**

### CODE REVIEW

**Comment by cockpituous** [comment]
Please manually check features related to these files:

```
package.json:    "xterm": "4.14.1"
pkg/lib/cockpit-components-terminal.jsx:import { Terminal as Term } from "xterm";
pkg/lib/console.css:@import "~xterm/css/xterm.css";
pkg/lib/console.css:.xterm-screen {
pkg/lib/context-menu.css:    /* xterm accessibility tree has z-index 100 and we need to be in front of it
pkg/systemd/terminal.jsx:                    "TERM=xterm-256color",
pkg/systemd/terminal.scss:.xterm-accessibility {
test/common/firefox-cdp-driver.js:        // HACK: Sometimes on Firefox >= 77 xterm.js can raise following message when Cockpit is reloaded:
test/verify/check-pages:            return '.terminal .xterm-accessibility-tree div:nth-child(%d)' % i
test/verify/check-pages:            b.wait_visible(".terminal .xterm-accessibility-tree")
test/verify/check-selinux:        b.wait_in_text(".xterm-accessibility-tree", "semanage import <<EOF")
test/verify/check-selinux:        b.wait_in_text(".xterm-accessibility-tree", "boolean -D")
test/verify/check-selinux:        b.wait_in_text(".xterm-accessibility-tree", "boolean -m -0 zebra_write_config")
test/verify/check-selinux:        b.wait_in_text(".xterm-accessibility-tree", "fcontext -a -f a -t samba_share_t -r 's0' '/var/tmp/'")
test/verify/check-shell-active-pages:            return '.terminal .xterm-accessibility-tree div:nth-child(%d)' % i
test/verify/check-shell-active-pages:        b.wait_visible(".terminal .xterm-accessibility-tree")
test/verify/check-system-realms:        b.wait_in_text(".terminal .xterm-accessibility-tree", "alice")
test/verify/check-system-realms:        b.wait_in_text(".terminal .xterm-accessibility-tree", "Ticket cache: FILE:/run/user")
test/verify/check-system-realms:        b.wait_in_text(".terminal .xterm-accessibility-tree", "Default principal: alice@COCKPIT.LAN")
test/verify/check-system-realms:        b.wait_in_text(".terminal .xterm-accessibility-tree", "for client HTTP/x0.cockpit.lan@COCKPIT.LAN")
test/verify/check-system-realms:        b.wait_in_text(".terminal .xterm-accessibility-tree", "admin")
test/verify/check-system-realms:        b.wait_in_text(".terminal .xterm-accessibility-tree", "Ticket cache")
test/verify/check-system-realms:        b.wait_in_text(".terminal .xterm-accessibility-tree", "Default principal: admin@COCKPIT.LAN")
test/verify/check-system-realms:        b.wait_in_text(".terminal .xterm-accessibility-tree", "krbtgt/COCKPIT.LAN")
test/verify/check-system-terminal:    return '.terminal .xterm-accessibility-tree div:nth-child(%d)' % i
test/verify/check-system-terminal:        b.wait_visible(".terminal .xterm-accessibility-tree")
test/verify/check-system-terminal:        b.wait_js_cond("ph_text('.xterm-accessibility-tree').indexOf('clear') < 0")
test/verify/check-system-terminal:            b.wait_in_text(".terminal .xterm-accessibility-tree", "9999989999991000000admin@")
test/verify/check-system-terminal:        b.assert_pixels("#terminal", "terminal", ignore=[".xterm-cursor-layer"])
test/verify/check-system-terminal:        b.wait_in_text(".terminal .xterm-accessibility-tree", "disconnected")
test/verify/check-system-terminal:        b.wait_not_in_text(".terminal .xterm-accessibility-tree", "disconnected")
test/verify/check-system-terminal:            b.wait_attr(".xterm-viewport", "style", style)
test/verify/check-system-terminal:        current_style = b.attr(".xterm-accessibility-tree div:first-child", "style")
test/verify/check-system-terminal:        new_style = b.attr(".xterm-accessibility-tree div:first-child", "style")
test/verify/check-system-terminal:        b.wait_attr(".xterm-viewport", "style", white_style)
test/verify/check-system-terminal:        b.wait_attr(".xterm-accessibility-tree div:first-child", "style", new_style)
test/verify/check-system-terminal:        b.wait_attr(".xterm-accessibility-tree div:first-child", "style", current_style)
test/verify/check-system-terminal:        b.wait_visible(".terminal .xterm-accessibility-tree")
```

**Comment by martinpitt** [comment]
[tests need updating](https://logs.cockpit-project.org/logs/pull-16340-20210913-082111-ed421fd8-debian-stable/log.html#273-2) to changed DOM structure of the terminal.

**Comment by KKoukiou** [comment]
> [tests need updating](https://logs.cockpit-project.org/logs/pull-16340-20210913-082111-ed421fd8-debian-stable/log.html#273-2) to changed DOM structure of the terminal.

I don't think it's that - the problem is that there is a width inline style as well on the .xterm-viewport.

**Comment by martinpitt** [comment]
Thanks @KKoukiou ! Still needswork, the pixel test [shows](https://logs.cockpit-project.org/logs/pull-16340-20210913-161426-9a18122e-fedora-34/TestTerminal-testBasic-terminal-pixels.png) that the scrollbar is broken. Yay for pixel tests!

**Comment by KKoukiou** [comment]
We have some other pixel tests PR open - let's get them first in and I will update the screenshots.
@garrett please take a look at the CSS change that it's fine.

**Review by garrett — CHANGES_REQUESTED** [review]
This fixes the scrollbar issue.

~However, the update breaks WebKit~:

Screenshot from 2021-09-14 15-47-29

(At first, I thought the problem might be related to fish shell, so I copy and pasted "bash" and still got the same issue.)

I'm going to check on main and in other versions of WebKit (as this is Canary), but meanwhile, I'll mark this as "request changes". If the same issue happens in main (just making sure this isn't a regression in xterm) and/or other WebKit versions are fine, then I'll change it to approve.

Update: Not a bug (except in Epiphany Canary, which is the unstable build of GNOME Web).

**Review by garrett — APPROVED** [review]
Duplicated character typing doesn't happen in stable or Nightly Epiphany on the branch. It also happens in Canary on the main branch. Therefore, this is a problem with Canary (which is "wild west" anyway), not with the update to the PR.

**Comment by garrett** [comment]
I'm normally not a fan of `!important`, but the widget hardcodes a pixel value which can run over the space for the widget, therefore, we had to override that with an `!important` to make it fit within the content given to it.

There did not appear to be a way to override the behavior of the widget directly, so the CSS override was necessary, so it wouldn't run out of bounds. It looks like it needs to now be applied to two classes instead of just one.

File: pkg/lib/console.css [hunk 1]
```diff
@@ -57,6 +57,6 @@
 }
 
 /* Ensure the console fits to its container (and doesn't attempt to go beyond the limits) */
```
Comment (garrett) [inline pkg/lib/console.css:59]: If we want to enhance the comment:

```suggestion
/* Ensure the console fits to its container (and doesn't attempt to go beyond the limits).
   This is needed as the widget hardcodes a pixel value for width that can be too wide. */
```

**Review by KKoukiou — APPROVED** [review]

**Review by martinpitt — APPROVED** [review]
Nice, thanks for sorting this out! Love the pixel tests.

The [new screenshot](https://github.com/cockpit-project/pixel-test-reference/blob/d93b68636e799b0dc32c1f213253424cd394f4e3/TestTerminal-testBasic-terminal-pixels.png) shows that the ends of the scrollbar are a wee bit off, but "good enough" at least for me.

**Comment by KKoukiou** [comment]
> Nice, thanks for sorting this out! Love the pixel tests.
> 
> The [new screenshot](https://github.com/cockpit-project/pixel-test-reference/blob/d93b68636e799b0dc32c1f213253424cd394f4e3/TestTerminal-testBasic-terminal-pixels.png) shows that the ends of the scrollbar are a wee bit off, but "good enough" at least for me.

@garrett wanna take a look and deal with it in a follow up? 
It will be again some  miracle CSS.

**Pull request merged**